Source/Extensions/Blazorise.TreeView/TreeView.razor.cs

```diff
@@ -11,21 +17,130 @@ public partial class TreeView<TNode> : BaseComponent
 {
     #region Members
 
-    private TreeViewState<TNode> state = new()
+    private _TreeViewNode<TNode> treeViewNodeRef;
+
+    private TreeViewState<TNode> treeViewState = new()
     {
+        SelectedNodes = new List<TNode>(),
+        SelectionMode = TreeViewSelectionMode.Single,
     };
 
+    private List<TreeViewNodeState<TNode>> treeViewNodeStates;
+
     #endregion
 
     #region Methods
 
+    public override async Task SetParametersAsync( ParameterView parameters )
+    {
+        bool nodesChanged = parameters.TryGetValue<IEnumerable<TNode>>( nameof( Nodes ), out var paramNodes ) && !paramNodes.AreEqual( Nodes );
+        bool selectedNodeChanged = parameters.TryGetValue<TNode>( nameof( SelectedNode ), out var paramSelectedNode ) && !paramSelectedNode.IsEqual( SelectedNode );
+        bool selectedNodesChanged = parameters.TryGetValue<IList<TNode>>( nameof( SelectedNodes ), out var paramSelectedNodes ) && !paramSelectedNodes.AreEqual( SelectedNodes );
+
+        if ( selectedNodeChanged )
+        {
+            treeViewState = treeViewState with { SelectedNode = paramSelectedNode };
+        }
+
+        if ( selectedNodesChanged )
+        {
+            treeViewState = treeViewState with { SelectedNodes = paramSelectedNodes };
+        }
+
+        await base.SetParametersAsync( parameters );
+
+        if ( nodesChanged )
+        {
+            treeViewNodeStates = new();
+
+            await foreach ( var nodeState in paramNodes.ToNodeStates( HasChildNodesAsync, HasChildNodes, ExpandedNodes.Intersect( paramNodes ?? Enumerable.Empty<TNode>() ).Any() ) )
+            {
+                treeViewNodeStates.Add( nodeState );
+            }
+        }
+
+        if ( selectedNodeChanged )
+        {
+            await SelectedNodeChanged.InvokeAsync( treeViewState.SelectedNode );
+        }
+
+        if ( selectedNodesChanged )
+        {
+            await SelectedNodesChanged.InvokeAsync( treeViewState.SelectedNodes );
+        }
+    }
+
+    protected override void BuildClasses( ClassBuilder builder )
+    {
+        builder.Append( "b-tree-view" );
+
+        base.BuildClasses( builder );
+    }
+
+    /// <summary>
+    /// Selects the node when in single selection mode.
+    /// </summary>
+    /// <param name="node">Node to select.</param>
     public void SelectNode( TNode node )

```

Comment:
Why not task?

Reply:
This is an existing API. We can do it in v2.